pkg/identity/gitlabcom/principal.go

```diff
@@ -104,6 +113,10 @@ func JobPrincipalFromIDToken(_ context.Context, token *oidc.IDToken) (identity.P
 		return nil, errors.New("missing pipeline_id claim in ID token")
 	}
 
+	if claims.PipelineRef == "" {
```

Comment:
Should Fulcio be okay with `pipeline_ref` being empty? For the first iteration, this claim is behind a feature flag and only populated it if the CI config lives in the same repo.

Reply:
if that is temporary i guess there is no problem. but not 100% sure to be honest

Reply:
👋  It seems preferable to error if this is not set, so we don't end up with partial signing certificates that lack this information. 

cc @haydentherapper thoughts on this?

Reply:
The problem with this is Build Signer URI would be empty (at least for now) which we document as `MUST` - https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#requirements-to-support-signing-with-cicd-workload-identities

Do we have an ETA on the upstream GitLab JWT changes landing? We might want to wait on those first before making this change.

Reply:
> Do we have an ETA on the upstream GitLab JWT changes landing? We might want to wait on those first before making this change.

@wlynch agreed, that was my intent. Sorry if that wasn't clear from the PR description. Hopefully we can get it in 16.1 without a feature flag, but if not then next month in 16.2.

There will still be some cases where the `pipeline_ref` cannot be populated after it is introduced. These cases probably represent <10% of real usage. We will continue to close that gap in subsequent iterations. For the time being, in situations where we cannot populate `pipeline_ref` on GitLab's side, I think it's better to fail than massage over it in Fulcio.

Thoughts?

Reply:
Agreed on erring out in Fulcio just in case. What is the action a user should take if they do see this error? Can we link them to documentation? Can they update their pipeline, or is there some other action?